# Analyse des données RH d’une entreprise

### Objectifs de la mission
#### 1. Charger les données avec Pandas

#### 2. Nettoyer les données (valeurs manquantes, doublons, formats)

#### 3. Analyser la structure de l’entreprise :

 - Nombre total d’employés

 - Répartition par département

 - Répartition H/F

 - Répartition par statut (CDI, CDD, Freelance, etc.)

#### 4. Statistiques descriptives :

 - Salaire moyen, médian, min, max

 - Âge moyen et ancienneté moyenne

 - Répartition des salaires par département

####  Chargement des librairies

In [66]:
import pandas as pd
import matplotlib.pyplot as mt
from datetime import datetime

#### 1. Charger les données avec Pandas

In [67]:
employees = pd.read_csv("/Users/Lenovo/Downloads/employes.csv")
employees.index = range(1, 101)
employees

Nom  Âge Sexe Département     Statut  Salaire (€) Date entrée
1      Tristan   24    F          IT  Stagiaire         3977  2015-09-02
2    Françoise   59    M          IT        CDI         3161  2024-02-21
3      Margaud   27    F  Commercial        CDI         3601  2019-11-26
4     Marcelle   53    M   Marketing  Stagiaire         3918  2023-12-17
5       Andrée   59    F          IT  Stagiaire         2445  2016-04-29
..         ...  ...  ...         ...        ...          ...         ...
96   Dominique   56    F   Marketing        CDI         2366  2016-07-01
97     Vincent   25    F          RH        CDD         5256  2023-09-28
98    Thibault   26    F     Finance  Stagiaire         5118  2021-04-14
99      Maggie   22    F   Marketing        CDI         3841  2023-01-17
100     Gérard   36    M          RH  Freelance         4772  2015-10-11

[100 rows x 7 columns]

In [68]:
De = list(employees["Date entrée"])
type(De[0])
#créer une colonne ancienneté

str

### Nous constatons que la colonne 'Date entrée' a toutes ses entrées en str( c'est à dire en chaîne de caractères), ce qui ne nous serait pas très utile pour ntre Analyse. Nous créerons donc une nouvelle colonne par la suite nommée "Ancienneté" qui contiendra l'âge de chaque employé.

#### 2. Nettoyer les données (valeurs manquantes, doublons, formats)

In [69]:
# ÉCRITURE DE LA FONCTION POUR LA COLONNE À AJOUTER
def transforming(Col) : 
    date_format = '%Y-%m-%d'
    new_Col = []
    for col in Col : 
        new_Col.append(datetime.strptime(col, date_format))
    return new_Col

In [70]:
#liste à utiliser
Date_entrée = transforming(employees['Date entrée'])
employees['Date entrée'] = Date_entrée

In [71]:
employees

Nom  Âge Sexe Département     Statut  Salaire (€) Date entrée
1      Tristan   24    F          IT  Stagiaire         3977  2015-09-02
2    Françoise   59    M          IT        CDI         3161  2024-02-21
3      Margaud   27    F  Commercial        CDI         3601  2019-11-26
4     Marcelle   53    M   Marketing  Stagiaire         3918  2023-12-17
5       Andrée   59    F          IT  Stagiaire         2445  2016-04-29
..         ...  ...  ...         ...        ...          ...         ...
96   Dominique   56    F   Marketing        CDI         2366  2016-07-01
97     Vincent   25    F          RH        CDD         5256  2023-09-28
98    Thibault   26    F     Finance  Stagiaire         5118  2021-04-14
99      Maggie   22    F   Marketing        CDI         3841  2023-01-17
100     Gérard   36    M          RH  Freelance         4772  2015-10-11

[100 rows x 7 columns]

#### Nous procédons maintenant à la créeation de la colonne "Ancienneté"

In [72]:
Date_act = datetime.now()
employees["Ancienneté (en jours)"] = [(Date_act - Date_en).days for Date_en in employees["Date entrée"]]
employees

Nom  Âge Sexe Département     Statut  Salaire (€) Date entrée  \
1      Tristan   24    F          IT  Stagiaire         3977  2015-09-02   
2    Françoise   59    M          IT        CDI         3161  2024-02-21   
3      Margaud   27    F  Commercial        CDI         3601  2019-11-26   
4     Marcelle   53    M   Marketing  Stagiaire         3918  2023-12-17   
5       Andrée   59    F          IT  Stagiaire         2445  2016-04-29   
..         ...  ...  ...         ...        ...          ...         ...   
96   Dominique   56    F   Marketing        CDI         2366  2016-07-01   
97     Vincent   25    F          RH        CDD         5256  2023-09-28   
98    Thibault   26    F     Finance  Stagiaire         5118  2021-04-14   
99      Maggie   22    F   Marketing        CDI         3841  2023-01-17   
100     Gérard   36    M          RH  Freelance         4772  2015-10-11   

     Ancienneté (en jours)  
1                     3598  
2                      504  
3                     2052  
4                      570  
5                     3358  
..                     ...  
96                    3295  
97                     650  
98                    1547  
99                     904  
100                   3559  

[100 rows x 8 columns]

In [73]:
employees.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 100 entries, 1 to 100
Data columns (total 8 columns):
 #   Column                 Non-Null Count  Dtype         
---  ------                 --------------  -----         
 0   Nom                    100 non-null    object        
 1   Âge                    100 non-null    int64         
 2   Sexe                   100 non-null    object        
 3   Département            100 non-null    object        
 4   Statut                 100 non-null    object        
 5   Salaire (€)            100 non-null    int64         
 6   Date entrée            100 non-null    datetime64[ns]
 7   Ancienneté (en jours)  100 non-null    int64         
dtypes: datetime64[ns](1), int64(3), object(4)
memory usage: 6.4+ KB


### Nous constatons que notre fichier ne contient aucune colonne sans valeur.

#### 3. Analyser la structure de l’entreprise :

 - Nombre total d’employés

In [74]:
taille = list(employees.shape)
effectif = taille[0]
print(f" Il y a excatement {effectif} employés.")

 Il y a excatement 100 employés.


- Répartition par département

In [75]:
# on veut connaître les noms des différents départements dans l'entreprise
départements = list(set(employees["Département"]))
count = 0
for dep in départements : 
    print(f" Département N°{count + 1} : {départements[count]}")
    count += 1

 Département N°1 : RH
 Département N°2 : Marketing
 Département N°3 : Finance
 Département N°4 : IT
 Département N°5 : Commercial


In [76]:
# Pour la répartition par département, nous créerons des sous data Frames en utilisant les différents noms
employees_finance = employees[employees.Département == "Finance"]
employees_Rh = employees[employees.Département == "RH"]
employees_Com = employees[employees.Département == "Commercial"]
employees_marketing = employees[employees.Département == "Marketing"]
employees_IT = employees[employees.Département == "IT"]

In [77]:
#Département Finance
employees_finance

Nom  Âge Sexe Département     Statut  Salaire (€) Date entrée  \
6   Marguerite   40    M     Finance  Stagiaire         4896  2018-03-05   
9   Frédérique   58    M     Finance  Stagiaire         3106  2025-04-09   
13    Philippe   57    M     Finance  Freelance         4267  2022-07-23   
20     William   44    F     Finance  Stagiaire         3014  2018-11-15   
26    Adélaïde   48    M     Finance  Stagiaire         1949  2017-06-07   
30      Alfred   45    F     Finance  Freelance         1977  2018-07-31   
33   Geneviève   46    M     Finance        CDD         4586  2017-01-12   
35      Élodie   55    M     Finance        CDD         3610  2022-11-02   
37      Émilie   27    F     Finance  Freelance         2489  2017-11-22   
44        Alix   43    M     Finance        CDD         5191  2022-11-27   
59      Julien   50    M     Finance        CDD         5833  2018-01-08   
62       Laure   45    M     Finance        CDI         5753  2017-05-05   
67    Margaret   54    F     Finance  Stagiaire         4396  2016-01-30   
81      Jérôme   31    F     Finance        CDI         5448  2017-03-19   
87       Émile   46    F     Finance        CDD         3810  2017-09-20   
88     Vincent   25    M     Finance  Freelance         2327  2016-11-21   
90      Maggie   38    M     Finance  Freelance         5131  2023-01-06   
92        Alix   49    M     Finance        CDD         4054  2022-08-19   
98    Thibault   26    F     Finance  Stagiaire         5118  2021-04-14   

    Ancienneté (en jours)  
6                    2683  
9                      91  
13                   1082  
20                   2428  
26                   2954  
30                   2535  
33                   3100  
35                    980  
37                   2786  
44                    955  
59                   2739  
62                   2987  
67                   3448  
81                   3034  
87                   2849  
88                   3152  
90                    915  
92                   1055  
98                   1547

In [78]:
#Département RH
employees_Rh

Nom  Âge Sexe Département     Statut  Salaire (€) Date entrée  \
8        Julie   43    F          RH        CDI         4754  2023-10-31   
14    Philippe   23    F          RH        CDI         4774  2023-01-07   
17      Julien   24    M          RH  Freelance         2517  2023-04-21   
19     Arnaude   41    F          RH  Freelance         5381  2023-06-22   
21      Michel   52    M          RH  Stagiaire         4329  2021-02-04   
32      Amélie   34    F          RH  Freelance         4215  2018-05-30   
36       Diane   28    M          RH  Stagiaire         5270  2017-05-27   
42      Martin   60    M          RH  Stagiaire         3398  2016-11-25   
45         Guy   35    F          RH        CDD         2621  2020-11-14   
46      Pierre   22    F          RH  Freelance         2263  2019-12-08   
49       Roger   27    F          RH  Stagiaire         1802  2025-04-23   
50      Andrée   42    M          RH        CDD         2410  2019-03-03   
57      Margot   23    M          RH        CDD         2749  2023-06-22   
58        Rémy   49    F          RH        CDI         2496  2016-04-28   
65     Valérie   35    M          RH        CDI         3107  2020-04-20   
79   Françoise   22    F          RH        CDD         3967  2018-08-01   
97     Vincent   25    F          RH        CDD         5256  2023-09-28   
100     Gérard   36    M          RH  Freelance         4772  2015-10-11   

     Ancienneté (en jours)  
8                      617  
14                     914  
17                     810  
19                     748  
21                    1616  
32                    2597  
36                    2965  
42                    3148  
45                    1698  
46                    2040  
49                      77  
50                    2320  
57                     748  
58                    3359  
65                    1906  
79                    2534  
97                     650  
100                   3559

In [79]:
#Département Commercial
employees_Com

Nom  Âge Sexe Département     Statut  Salaire (€) Date entrée  \
3     Margaud   27    F  Commercial        CDI         3601  2019-11-26   
27      Diane   43    M  Commercial        CDD         5431  2017-07-19   
31    Vincent   54    M  Commercial        CDI         5651  2019-01-19   
38       Jean   57    M  Commercial  Freelance         5530  2020-01-25   
51  Christine   24    M  Commercial  Stagiaire         4857  2022-07-23   
60    Thérèse   24    M  Commercial        CDI         2180  2018-06-04   
61   Zacharie   30    F  Commercial  Stagiaire         3612  2016-07-20   
63      Anaïs   24    F  Commercial        CDI         1888  2022-01-17   
68   Brigitte   32    F  Commercial        CDI         2297  2020-09-02   
72       Marc   42    F  Commercial  Stagiaire         2473  2024-02-20   
73   Matthieu   27    F  Commercial  Freelance         3584  2015-11-22   
93       Nath   24    M  Commercial  Stagiaire         2711  2016-01-07   
95    Tristan   60    M  Commercial        CDI         2075  2017-11-14   

    Ancienneté (en jours)  
3                    2052  
27                   2912  
31                   2363  
38                   1992  
51                   1082  
60                   2592  
61                   3276  
63                   1269  
68                   1771  
72                    505  
73                   3517  
93                   3471  
95                   2794

In [80]:
#Département Marketing
employees_marketing

Nom  Âge Sexe Département     Statut  Salaire (€) Date entrée  \
4     Marcelle   53    M   Marketing  Stagiaire         3918  2023-12-17   
7     Laetitia   24    F   Marketing  Freelance         5004  2018-10-19   
12  Frédérique   33    F   Marketing        CDI         4616  2023-11-08   
18       Anaïs   44    M   Marketing        CDD         2550  2022-06-25   
23      Andrée   41    M   Marketing  Stagiaire         4093  2021-05-31   
34  Antoinette   43    F   Marketing  Freelance         1867  2016-06-04   
40      Hugues   37    M   Marketing        CDD         2876  2020-02-22   
43    Bertrand   27    F   Marketing        CDI         5562  2018-01-24   
48    Victoire   29    F   Marketing        CDI         5861  2018-02-11   
54      Benoît   57    F   Marketing  Stagiaire         2628  2020-01-16   
66     Maurice   22    F   Marketing  Freelance         4865  2016-08-05   
69     Gilbert   28    M   Marketing        CDI         5341  2016-07-27   
71      Sylvie   57    M   Marketing  Freelance         3192  2025-05-24   
74     Patrick   46    F   Marketing        CDI         4459  2020-12-05   
75     Laurent   58    M   Marketing        CDI         2044  2023-10-30   
76    Patricia   32    F   Marketing  Stagiaire         5865  2024-06-06   
77    Virginie   27    F   Marketing        CDI         1856  2016-01-14   
82       Henri   26    F   Marketing        CDI         5519  2025-02-27   
86       André   40    F   Marketing        CDD         3029  2025-01-10   
96   Dominique   56    F   Marketing        CDI         2366  2016-07-01   
99      Maggie   22    F   Marketing        CDI         3841  2023-01-17   

    Ancienneté (en jours)  
4                     570  
7                    2455  
12                    609  
18                   1110  
23                   1500  
34                   3322  
40                   1964  
43                   2723  
48                   2705  
54                   2001  
66                   3260  
69                   3269  
71                     46  
74                   1677  
75                    618  
76                    398  
77                   3464  
82                    132  
86                    180  
96                   3295  
99                    904

In [81]:
#Département IT
employees_IT

Nom  Âge Sexe Département     Statut  Salaire (€) Date entrée  \
1      Tristan   24    F          IT  Stagiaire         3977  2015-09-02   
2    Françoise   59    M          IT        CDI         3161  2024-02-21   
5       Andrée   59    F          IT  Stagiaire         2445  2016-04-29   
10  Antoinette   36    F          IT        CDI         2415  2017-08-31   
11      Odette   60    F          IT        CDD         3591  2017-01-30   
15       Renée   42    F          IT  Freelance         2566  2021-01-05   
16    Lorraine   46    F          IT        CDD         3520  2024-10-23   
22    Mathilde   41    F          IT  Freelance         4232  2025-05-16   
24    Stéphane   56    F          IT        CDD         1935  2021-06-24   
25      Joseph   26    F          IT  Freelance         1957  2024-01-28   
28    Lorraine   48    M          IT        CDD         5621  2024-08-21   
29        Alex   32    M          IT        CDD         2743  2021-02-10   
39      Xavier   51    F          IT  Stagiaire         4628  2019-02-13   
41       Roger   54    F          IT        CDD         5771  2018-04-05   
47  Alexandria   43    F          IT  Freelance         5347  2023-06-24   
52     Martine   45    M          IT  Freelance         4251  2023-06-03   
53   Élisabeth   48    F          IT  Freelance         2869  2018-12-06   
55      Joseph   50    F          IT  Stagiaire         5506  2025-04-01   
56     Arnaude   57    F          IT  Freelance         2099  2021-12-08   
64    Michelle   35    F          IT        CDD         5856  2017-09-16   
70    Marcelle   24    M          IT        CDD         5042  2016-03-22   
78   Véronique   36    M          IT        CDI         5822  2022-05-20   
80     Michèle   22    M          IT        CDI         3607  2019-12-24   
83     Olivier   60    F          IT        CDD         3396  2018-11-03   
84    François   55    M          IT        CDI         3906  2017-03-31   
85      Hugues   28    F          IT  Freelance         4101  2024-03-22   
89      Amélie   48    F          IT        CDI         4865  2017-09-24   
91      Margot   33    F          IT  Stagiaire         4850  2023-12-07   
94      Julien   27    F          IT  Stagiaire         5211  2018-12-05   

    Ancienneté (en jours)  
1                    3598  
2                     504  
5                    3358  
10                   2869  
11                   3082  
15                   1646  
16                    259  
22                     54  
24                   1476  
25                    528  
28                    322  
29                   1610  
39                   2338  
41                   2652  
47                    746  
52                    767  
53                   2407  
55                     99  
56                   1309  
64                   2853  
70                   3396  
78                   1146  
80                   2024  
83                   2440  
84                   3022  
85                    474  
89                   2845  
91                    580  
94                   2408

- Répartition H/F

In [82]:
employees_hommes = employees[employees.Sexe == "M"]
employees_femmes = employees[employees.Sexe == "F"]

In [83]:
# Les hommes
employees_hommes

Nom  Âge Sexe Département     Statut  Salaire (€) Date entrée  \
2     Françoise   59    M          IT        CDI         3161  2024-02-21   
4      Marcelle   53    M   Marketing  Stagiaire         3918  2023-12-17   
6    Marguerite   40    M     Finance  Stagiaire         4896  2018-03-05   
9    Frédérique   58    M     Finance  Stagiaire         3106  2025-04-09   
13     Philippe   57    M     Finance  Freelance         4267  2022-07-23   
17       Julien   24    M          RH  Freelance         2517  2023-04-21   
18        Anaïs   44    M   Marketing        CDD         2550  2022-06-25   
21       Michel   52    M          RH  Stagiaire         4329  2021-02-04   
23       Andrée   41    M   Marketing  Stagiaire         4093  2021-05-31   
26     Adélaïde   48    M     Finance  Stagiaire         1949  2017-06-07   
27        Diane   43    M  Commercial        CDD         5431  2017-07-19   
28     Lorraine   48    M          IT        CDD         5621  2024-08-21   
29         Alex   32    M          IT        CDD         2743  2021-02-10   
31      Vincent   54    M  Commercial        CDI         5651  2019-01-19   
33    Geneviève   46    M     Finance        CDD         4586  2017-01-12   
35       Élodie   55    M     Finance        CDD         3610  2022-11-02   
36        Diane   28    M          RH  Stagiaire         5270  2017-05-27   
38         Jean   57    M  Commercial  Freelance         5530  2020-01-25   
40       Hugues   37    M   Marketing        CDD         2876  2020-02-22   
42       Martin   60    M          RH  Stagiaire         3398  2016-11-25   
44         Alix   43    M     Finance        CDD         5191  2022-11-27   
50       Andrée   42    M          RH        CDD         2410  2019-03-03   
51    Christine   24    M  Commercial  Stagiaire         4857  2022-07-23   
52      Martine   45    M          IT  Freelance         4251  2023-06-03   
57       Margot   23    M          RH        CDD         2749  2023-06-22   
59       Julien   50    M     Finance        CDD         5833  2018-01-08   
60      Thérèse   24    M  Commercial        CDI         2180  2018-06-04   
62        Laure   45    M     Finance        CDI         5753  2017-05-05   
65      Valérie   35    M          RH        CDI         3107  2020-04-20   
69      Gilbert   28    M   Marketing        CDI         5341  2016-07-27   
70     Marcelle   24    M          IT        CDD         5042  2016-03-22   
71       Sylvie   57    M   Marketing  Freelance         3192  2025-05-24   
75      Laurent   58    M   Marketing        CDI         2044  2023-10-30   
78    Véronique   36    M          IT        CDI         5822  2022-05-20   
80      Michèle   22    M          IT        CDI         3607  2019-12-24   
84     François   55    M          IT        CDI         3906  2017-03-31   
88      Vincent   25    M     Finance  Freelance         2327  2016-11-21   
90       Maggie   38    M     Finance  Freelance         5131  2023-01-06   
92         Alix   49    M     Finance        CDD         4054  2022-08-19   
93         Nath   24    M  Commercial  Stagiaire         2711  2016-01-07   
95      Tristan   60    M  Commercial        CDI         2075  2017-11-14   
100      Gérard   36    M          RH  Freelance         4772  2015-10-11   

     Ancienneté (en jours)  
2                      504  
4                      570  
6                     2683  
9                       91  
13                    1082  
17                     810  
18                    1110  
21                    1616  
23                    1500  
26                    2954  
27                    2912  
28                     322  
29                    1610  
31                    2363  
33                    3100  
35                     980  
36                    2965  
38                    1992  
40                    1964  
42                    3148  
44                     955  
50                    2320  
51                    1082  
52  

In [84]:
#Les femmes
employees_femmes

Nom  Âge Sexe Département     Statut  Salaire (€) Date entrée  \
1      Tristan   24    F          IT  Stagiaire         3977  2015-09-02   
3      Margaud   27    F  Commercial        CDI         3601  2019-11-26   
5       Andrée   59    F          IT  Stagiaire         2445  2016-04-29   
7     Laetitia   24    F   Marketing  Freelance         5004  2018-10-19   
8        Julie   43    F          RH        CDI         4754  2023-10-31   
10  Antoinette   36    F          IT        CDI         2415  2017-08-31   
11      Odette   60    F          IT        CDD         3591  2017-01-30   
12  Frédérique   33    F   Marketing        CDI         4616  2023-11-08   
14    Philippe   23    F          RH        CDI         4774  2023-01-07   
15       Renée   42    F          IT  Freelance         2566  2021-01-05   
16    Lorraine   46    F          IT        CDD         3520  2024-10-23   
19     Arnaude   41    F          RH  Freelance         5381  2023-06-22   
20     William   44    F     Finance  Stagiaire         3014  2018-11-15   
22    Mathilde   41    F          IT  Freelance         4232  2025-05-16   
24    Stéphane   56    F          IT        CDD         1935  2021-06-24   
25      Joseph   26    F          IT  Freelance         1957  2024-01-28   
30      Alfred   45    F     Finance  Freelance         1977  2018-07-31   
32      Amélie   34    F          RH  Freelance         4215  2018-05-30   
34  Antoinette   43    F   Marketing  Freelance         1867  2016-06-04   
37      Émilie   27    F     Finance  Freelance         2489  2017-11-22   
39      Xavier   51    F          IT  Stagiaire         4628  2019-02-13   
41       Roger   54    F          IT        CDD         5771  2018-04-05   
43    Bertrand   27    F   Marketing        CDI         5562  2018-01-24   
45         Guy   35    F          RH        CDD         2621  2020-11-14   
46      Pierre   22    F          RH  Freelance         2263  2019-12-08   
47  Alexandria   43    F          IT  Freelance         5347  2023-06-24   
48    Victoire   29    F   Marketing        CDI         5861  2018-02-11   
49       Roger   27    F          RH  Stagiaire         1802  2025-04-23   
53   Élisabeth   48    F          IT  Freelance         2869  2018-12-06   
54      Benoît   57    F   Marketing  Stagiaire         2628  2020-01-16   
55      Joseph   50    F          IT  Stagiaire         5506  2025-04-01   
56     Arnaude   57    F          IT  Freelance         2099  2021-12-08   
58        Rémy   49    F          RH        CDI         2496  2016-04-28   
61    Zacharie   30    F  Commercial  Stagiaire         3612  2016-07-20   
63       Anaïs   24    F  Commercial        CDI         1888  2022-01-17   
64    Michelle   35    F          IT        CDD         5856  2017-09-16   
66     Maurice   22    F   Marketing  Freelance         4865  2016-08-05   
67    Margaret   54    F     Finance  Stagiaire         4396  2016-01-30   
68    Brigitte   32    F  Commercial        CDI         2297  2020-09-02   
72        Marc   42    F  Commercial  Stagiaire         2473  2024-02-20   
73    Matthieu   27    F  Commercial  Freelance         3584  2015-11-22   
74     Patrick   46    F   Marketing        CDI         4459  2020-12-05   
76    Patricia   32    F   Marketing  Stagiaire         5865  2024-06-06   
77    Virginie   27    F   Marketing        CDI         1856  2016-01-14   
79   Françoise   22    F          RH        CDD         3967  2018-08-01   
81      Jérôme   31    F     Finance        CDI         5448  2017-03-19   
82       Henri   26    F   Marketing        CDI         5519  2025-02-27   
83     Olivier   60    F          IT        CDD         3396  2018-11-03   
85      Hugues   28    F          IT  Freelance         4101  2024-03-22   
86       André   40    F   Marketing        CDD         3029  2025-01-10   
87       Émile   46    F     Finance        CDD         3810  2017-09-20   
89      Amélie   48    F          IT        CDI         486

- Répartition par statut (CDI, CDD, Freelance, etc.)

In [85]:
# Voyons quels sont les différents statuts
statuts = list(set(employees.Statut))
counting = 0
for statut in statuts : 
    print(f"Statut N°{counting + 1} : {statuts[counting]}")
    counting += 1

Statut N°1 : CDD
Statut N°2 : CDI
Statut N°3 : Freelance
Statut N°4 : Stagiaire


In [86]:
#Nous avons ainsi 4 statuts dans l'entreprise
#Procédons maintenant à la répartition sous forme de différents tableaux
employees_Freelance = employees[employees.Statut == "Freelance"]
employees_Sta = employees[employees.Statut == "Stagiaire"]
employees_CDD = employees[employees.Statut == "CDD"]
employees_CDI = employees[employees.Statut == "CDI"]

In [87]:
#Freelance
employees_Freelance

Nom  Âge Sexe Département     Statut  Salaire (€) Date entrée  \
7      Laetitia   24    F   Marketing  Freelance         5004  2018-10-19   
13     Philippe   57    M     Finance  Freelance         4267  2022-07-23   
15        Renée   42    F          IT  Freelance         2566  2021-01-05   
17       Julien   24    M          RH  Freelance         2517  2023-04-21   
19      Arnaude   41    F          RH  Freelance         5381  2023-06-22   
22     Mathilde   41    F          IT  Freelance         4232  2025-05-16   
25       Joseph   26    F          IT  Freelance         1957  2024-01-28   
30       Alfred   45    F     Finance  Freelance         1977  2018-07-31   
32       Amélie   34    F          RH  Freelance         4215  2018-05-30   
34   Antoinette   43    F   Marketing  Freelance         1867  2016-06-04   
37       Émilie   27    F     Finance  Freelance         2489  2017-11-22   
38         Jean   57    M  Commercial  Freelance         5530  2020-01-25   
46       Pierre   22    F          RH  Freelance         2263  2019-12-08   
47   Alexandria   43    F          IT  Freelance         5347  2023-06-24   
52      Martine   45    M          IT  Freelance         4251  2023-06-03   
53    Élisabeth   48    F          IT  Freelance         2869  2018-12-06   
56      Arnaude   57    F          IT  Freelance         2099  2021-12-08   
66      Maurice   22    F   Marketing  Freelance         4865  2016-08-05   
71       Sylvie   57    M   Marketing  Freelance         3192  2025-05-24   
73     Matthieu   27    F  Commercial  Freelance         3584  2015-11-22   
85       Hugues   28    F          IT  Freelance         4101  2024-03-22   
88      Vincent   25    M     Finance  Freelance         2327  2016-11-21   
90       Maggie   38    M     Finance  Freelance         5131  2023-01-06   
100      Gérard   36    M          RH  Freelance         4772  2015-10-11   

     Ancienneté (en jours)  
7                     2455  
13                    1082  
15                    1646  
17                     810  
19                     748  
22                      54  
25                     528  
30                    2535  
32                    2597  
34                    3322  
37                    2786  
38                    1992  
46                    2040  
47                     746  
52                     767  
53                    2407  
56                    1309  
66                    3260  
71                      46  
73                    3517  
85                     474  
88                    3152  
90                     915  
100                   3559

In [88]:
#Stagiaire
employees_Sta

Nom  Âge Sexe Département     Statut  Salaire (€) Date entrée  \
1      Tristan   24    F          IT  Stagiaire         3977  2015-09-02   
4     Marcelle   53    M   Marketing  Stagiaire         3918  2023-12-17   
5       Andrée   59    F          IT  Stagiaire         2445  2016-04-29   
6   Marguerite   40    M     Finance  Stagiaire         4896  2018-03-05   
9   Frédérique   58    M     Finance  Stagiaire         3106  2025-04-09   
20     William   44    F     Finance  Stagiaire         3014  2018-11-15   
21      Michel   52    M          RH  Stagiaire         4329  2021-02-04   
23      Andrée   41    M   Marketing  Stagiaire         4093  2021-05-31   
26    Adélaïde   48    M     Finance  Stagiaire         1949  2017-06-07   
36       Diane   28    M          RH  Stagiaire         5270  2017-05-27   
39      Xavier   51    F          IT  Stagiaire         4628  2019-02-13   
42      Martin   60    M          RH  Stagiaire         3398  2016-11-25   
49       Roger   27    F          RH  Stagiaire         1802  2025-04-23   
51   Christine   24    M  Commercial  Stagiaire         4857  2022-07-23   
54      Benoît   57    F   Marketing  Stagiaire         2628  2020-01-16   
55      Joseph   50    F          IT  Stagiaire         5506  2025-04-01   
61    Zacharie   30    F  Commercial  Stagiaire         3612  2016-07-20   
67    Margaret   54    F     Finance  Stagiaire         4396  2016-01-30   
72        Marc   42    F  Commercial  Stagiaire         2473  2024-02-20   
76    Patricia   32    F   Marketing  Stagiaire         5865  2024-06-06   
91      Margot   33    F          IT  Stagiaire         4850  2023-12-07   
93        Nath   24    M  Commercial  Stagiaire         2711  2016-01-07   
94      Julien   27    F          IT  Stagiaire         5211  2018-12-05   
98    Thibault   26    F     Finance  Stagiaire         5118  2021-04-14   

    Ancienneté (en jours)  
1                    3598  
4                     570  
5                    3358  
6                    2683  
9                      91  
20                   2428  
21                   1616  
23                   1500  
26                   2954  
36                   2965  
39                   2338  
42                   3148  
49                     77  
51                   1082  
54                   2001  
55                     99  
61                   3276  
67                   3448  
72                    505  
76                    398  
91                    580  
93                   3471  
94                   2408  
98                   1547

In [89]:
#CDD
employees_CDD

Nom  Âge Sexe Département Statut  Salaire (€) Date entrée  \
11     Odette   60    F          IT    CDD         3591  2017-01-30   
16   Lorraine   46    F          IT    CDD         3520  2024-10-23   
18      Anaïs   44    M   Marketing    CDD         2550  2022-06-25   
24   Stéphane   56    F          IT    CDD         1935  2021-06-24   
27      Diane   43    M  Commercial    CDD         5431  2017-07-19   
28   Lorraine   48    M          IT    CDD         5621  2024-08-21   
29       Alex   32    M          IT    CDD         2743  2021-02-10   
33  Geneviève   46    M     Finance    CDD         4586  2017-01-12   
35     Élodie   55    M     Finance    CDD         3610  2022-11-02   
40     Hugues   37    M   Marketing    CDD         2876  2020-02-22   
41      Roger   54    F          IT    CDD         5771  2018-04-05   
44       Alix   43    M     Finance    CDD         5191  2022-11-27   
45        Guy   35    F          RH    CDD         2621  2020-11-14   
50     Andrée   42    M          RH    CDD         2410  2019-03-03   
57     Margot   23    M          RH    CDD         2749  2023-06-22   
59     Julien   50    M     Finance    CDD         5833  2018-01-08   
64   Michelle   35    F          IT    CDD         5856  2017-09-16   
70   Marcelle   24    M          IT    CDD         5042  2016-03-22   
79  Françoise   22    F          RH    CDD         3967  2018-08-01   
83    Olivier   60    F          IT    CDD         3396  2018-11-03   
86      André   40    F   Marketing    CDD         3029  2025-01-10   
87      Émile   46    F     Finance    CDD         3810  2017-09-20   
92       Alix   49    M     Finance    CDD         4054  2022-08-19   
97    Vincent   25    F          RH    CDD         5256  2023-09-28   

    Ancienneté (en jours)  
11                   3082  
16                    259  
18                   1110  
24                   1476  
27                   2912  
28                    322  
29                   1610  
33                   3100  
35                    980  
40                   1964  
41                   2652  
44                    955  
45                   1698  
50                   2320  
57                    748  
59                   2739  
64                   2853  
70                   3396  
79                   2534  
83                   2440  
86                    180  
87                   2849  
92                   1055  
97                    650

In [90]:
#CDI
employees_CDI

Nom  Âge Sexe Département Statut  Salaire (€) Date entrée  \
2    Françoise   59    M          IT    CDI         3161  2024-02-21   
3      Margaud   27    F  Commercial    CDI         3601  2019-11-26   
8        Julie   43    F          RH    CDI         4754  2023-10-31   
10  Antoinette   36    F          IT    CDI         2415  2017-08-31   
12  Frédérique   33    F   Marketing    CDI         4616  2023-11-08   
14    Philippe   23    F          RH    CDI         4774  2023-01-07   
31     Vincent   54    M  Commercial    CDI         5651  2019-01-19   
43    Bertrand   27    F   Marketing    CDI         5562  2018-01-24   
48    Victoire   29    F   Marketing    CDI         5861  2018-02-11   
58        Rémy   49    F          RH    CDI         2496  2016-04-28   
60     Thérèse   24    M  Commercial    CDI         2180  2018-06-04   
62       Laure   45    M     Finance    CDI         5753  2017-05-05   
63       Anaïs   24    F  Commercial    CDI         1888  2022-01-17   
65     Valérie   35    M          RH    CDI         3107  2020-04-20   
68    Brigitte   32    F  Commercial    CDI         2297  2020-09-02   
69     Gilbert   28    M   Marketing    CDI         5341  2016-07-27   
74     Patrick   46    F   Marketing    CDI         4459  2020-12-05   
75     Laurent   58    M   Marketing    CDI         2044  2023-10-30   
77    Virginie   27    F   Marketing    CDI         1856  2016-01-14   
78   Véronique   36    M          IT    CDI         5822  2022-05-20   
80     Michèle   22    M          IT    CDI         3607  2019-12-24   
81      Jérôme   31    F     Finance    CDI         5448  2017-03-19   
82       Henri   26    F   Marketing    CDI         5519  2025-02-27   
84    François   55    M          IT    CDI         3906  2017-03-31   
89      Amélie   48    F          IT    CDI         4865  2017-09-24   
95     Tristan   60    M  Commercial    CDI         2075  2017-11-14   
96   Dominique   56    F   Marketing    CDI         2366  2016-07-01   
99      Maggie   22    F   Marketing    CDI         3841  2023-01-17   

    Ancienneté (en jours)  
2                     504  
3                    2052  
8                     617  
10                   2869  
12                    609  
14                    914  
31                   2363  
43                   2723  
48                   2705  
58                   3359  
60                   2592  
62                   2987  
63                   1269  
65                   1906  
68                   1771  
69                   3269  
74                   1677  
75                    618  
77                   3464  
78                   1146  
80                   2024  
81                   3034  
82                    132  
84                   3022  
89                   2845  
95                   2794  
96                   3295  
99                    904

#### 4. Statistiques descriptives :

 - Salaire moyen, médian, min, max

In [91]:
salaires_df = employees[['Nom', 'Salaire (€)']]
salaires_df

Nom  Salaire (€)
1      Tristan         3977
2    Françoise         3161
3      Margaud         3601
4     Marcelle         3918
5       Andrée         2445
..         ...          ...
96   Dominique         2366
97     Vincent         5256
98    Thibault         5118
99      Maggie         3841
100     Gérard         4772

[100 rows x 2 columns]

In [92]:
salaires_df.describe() # ce tableau contient toutes les informations demandées.

Salaire (€)
count   100.000000
mean   3855.680000
std    1284.486942
min    1802.000000
25%    2607.250000
50%    3912.000000
75%    5013.500000
max    5865.000000

- Âge moyen et ancienneté moyenne

In [93]:
age = list(employees['Âge'])
ancienneté = list(employees['Ancienneté (en jours)'])

In [94]:
#Âge moyen
print(f"L'âge moyen est de {round(sum(age)/len(age), 2)} ans")

L'âge moyen est de 39.63 ans


In [95]:
print(f"L'ancienneté moyenne est de {round(sum(ancienneté)/len(ancienneté), 2)} jours")

L'ancienneté moyenne est de 1902.36 jours


- Répartition des salaires par département

In [96]:
#Département Finance
pd.DataFrame(employees_finance['Salaire (€)'])

Salaire (€)
6          4896
9          3106
13         4267
20         3014
26         1949
30         1977
33         4586
35         3610
37         2489
44         5191
59         5833
62         5753
67         4396
81         5448
87         3810
88         2327
90         5131
92         4054
98         5118

In [97]:
#Département IT
pd.DataFrame(employees_IT['Salaire (€)'])

Salaire (€)
1          3977
2          3161
5          2445
10         2415
11         3591
15         2566
16         3520
22         4232
24         1935
25         1957
28         5621
29         2743
39         4628
41         5771
47         5347
52         4251
53         2869
55         5506
56         2099
64         5856
70         5042
78         5822
80         3607
83         3396
84         3906
85         4101
89         4865
91         4850
94         5211

In [98]:
#Département Commercial
pd.DataFrame(employees_Com['Salaire (€)'])

Salaire (€)
3          3601
27         5431
31         5651
38         5530
51         4857
60         2180
61         3612
63         1888
68         2297
72         2473
73         3584
93         2711
95         2075

In [99]:
#département RH
pd.DataFrame(employees_Rh['Salaire (€)'])

Salaire (€)
8           4754
14          4774
17          2517
19          5381
21          4329
32          4215
36          5270
42          3398
45          2621
46          2263
49          1802
50          2410
57          2749
58          2496
65          3107
79          3967
97          5256
100         4772

In [100]:
#département Marketing
pd.DataFrame(employees_marketing['Salaire (€)'])

Salaire (€)
4          3918
7          5004
12         4616
18         2550
23         4093
34         1867
40         2876
43         5562
48         5861
54         2628
66         4865
69         5341
71         3192
74         4459
75         2044
76         5865
77         1856
82         5519
86         3029
96         2366
99         3841

### Nous constatons que le département avec le salaire le plus élevé est le département marketing, un salaire de 5865 €
### Une analyse bien plus profonde peut être faite, mais nous nous arrêterons à cette analyse purement descriptive.